charts/hedgedoc/templates/deployment.yaml

```
@@ -0,0 +1,70 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "hedgedoc.fullname" . }}
+  labels:
+    {{- include "hedgedoc.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "hedgedoc.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "hedgedoc.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "hedgedoc.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          env:
+            {{- range $key, $val := .Values.hedgedoc.env }}
+            - name: {{ $key }}
+              value: {{ $val | quote }}
```

for templating vaulues to work you would need to call the tpl function
```suggestion
              value: {{ tpl $val | quote }}
```
in this case, i wouldn't recommend that tho, i'd recommend removing the port env all together and doing something like this:

```yaml
            - name: "CMD_PORT"
              value: {{ .Values.service.main.ports.main.port | quote }}
            {{- range $key, $val := .Values.hedgedoc.env }}
            - name: {{ $key }}
              value: {{ $val | quote }}
            {{- end }}

```

---

:cook: :kiss: